notebooks/tel_and_site/subsys_req_ver/m1m3/SITCOM-989_analysis_inertia_compensation.ipynb

```diff
@@ -0,0 +1,741 @@
+    "    t_pad : int, default=1\n",
+    "        Time padding in seconds to be added at the beginning and at the end of a slew event.\n",
+    "\n",
+    "    Returns\n",
+    "    -------\n",
+    "    list\n",
+    "        A list of TMA slew events that occurred within the specified time range.\n",
+    "\n",
+    "    Notes\n",
+    "    -----\n",
+    "    This function retrieves TMA slew events occurring between the specified start and end times.\n",
+    "    It uses the TMAEventMaker class to obtain events for the specified day of observation (dayObs).\n",
+    "    The events are filtered to include only those that start after 1 second before the specified start time\n",
+    "    and end before 1 second after the specified end time.\n",
+    "\n",
+    "    Example\n",
+    "    -------\n",
+    "    >>> start_time = \"2023-08-09T00:00:00\"\n",
+    "    >>> end_time = \"2023-08-09T23:59:59\"\n",
+    "    >>> tma_events = get_tma_slew_events(start_time, end_time)\n",
+    "    \"\"\"\n",
+    "    dayObs = getDayObsForTime(dataset.begin)\n",
+    "    print(dataset.begin, dayObs)\n",
+    "    \n",
+    "    eventMaker = TMAEventMaker()\n",
+    "    events = eventMaker.getEvents(dayObs)\n",
+    "    \n",
+    "    events = [e for e in events if (e.begin > dataset.begin - TimeDelta(t_pad, format=\"sec\"))\n",
+    "              & ((e.end < dataset.end + TimeDelta(t_pad, format=\"sec\")))]\n",
+    "    \n",
+    "    return events"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "49cde41d-a015-482d-90be-a62368b45d2f",
+   "metadata": {},
+   "source": [
+    "## Data Collection"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "5c510e63-2ef5-4480-a7ff-990749fee22d",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "# Dataset 1 - Az Slew from -100 to 100 with 80 El, 30% motion settings and Inertia Compensation On\n",
+    "begin = \"2023-08-02T22:02:30\"\n",
+    "end = \"2023-08-02T22:04:00\"\n",
+    "\n",
+    "dataset1 = SlewDataSet(\n",
+    "    name=\"Az Slew from -100 to 100 with 80 El, 30% motion settings and Inertia Compensation On\",\n",
+    "    begin=Time(begin, format=\"isot\", scale=\"utc\"), \n",
+    "    end=Time(end, format=\"isot\", scale=\"utc\"),\n",
+    ")\n",
+    "\n",
+    "dataset1.df = await query_dataset(dataset1.begin, dataset1.end)\n",
+    "dataset1.events = get_tma_slew_events(dataset1)\n",
+    "dataset1.stats = get_metrics(dataset1)\n",
+    "dataset1.integrals = get_integral(dataset1)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "974c2eae-751d-4d85-b115-ca34452f7664",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "# Dataset 2 - Az Slew from -100 to 100 with 80 El, 30% motion settings and Inertia Compensation On\n",
+    "begin = \"2023-07-28T02:20:00\"\n",
```

Comment:
begin and end time start shall be swapped - this returns empty dataframe, which causes problems in the code.

Reply:
Good catch! Thanks, I fixed it